packages/tpetra/core/src/Tpetra_Details_EquilibrationInfo.hpp

```diff
@@ -138,21 +138,27 @@ struct EquilibrationInfo {
   void
   assign (const EquilibrationInfo<ScalarType, SrcDeviceType>& src)
   {
+  std::terminate();
+    // DEEP_COPY REVIEW - DEVICE-TO-DEVICE
     Kokkos::deep_copy (rowNorms, src.rowNorms);
```

Comment:
These guys are probably device-to-device in the tests, but the code is written to allow the device_type and the SrcDeviceType to be different.

Ugh.

Admittedly, we might not *need* this functionality, so if that's true we could rip it the SrcDeviceType stuff.

I vote for deferring a decision on this guy until we get feedback from the Kokkos team on Geoff's test.
I want to know what 3-arg is supposed to do if the views are in different memory spaces.  If it doesn't cost us much to keep the functionality, we can keep it.  Otherwise we can yank it.